src/coreclr/vm/comdelegate.cpp

```diff
@@ -330,6 +339,89 @@ BOOL AddNextShuffleEntryToArray(ArgLocDesc sArgSrc, ArgLocDesc sArgDst, SArray<S
         _ASSERTE(iteratorDst.HasNextOfs());
+#if defined(TARGET_LOONGARCH64)
+        if (!sArgDst.m_byteStackSize && (sArgSrc.m_offs > 0))
+        {
+            //assert(sArgSrc.m_idxStack >= 0);
+            if (sArgSrc.m_byteStackSize > 0)
+            {
+                if (sArgSrc.m_cGenReg > 0)
+                {
+                    assert(!"---------No this case--------------");
+                    assert(sArgSrc.m_cGenReg == 1);
+                    assert(sArgSrc.m_offs == 1);
+                }
+
+                assert(sArgDst.m_cFloatReg > 0);
+
+                if ((sArgSrc.m_offs == 1) || (sArgSrc.m_offs == 4))
+                {//first float;   ||   second float;
+                    //assert((sArgDst.m_cFloatReg + sArgDst.m_cGenReg) == sArgDst.m_cFloatReg);
+                    assert(sArgDst.m_cGenReg == 1);
+                    assert(sArgDst.m_cFloatReg == 1);
+                    assert(sArgSrc.m_byteStackSize == 8);
+
+                    entry.srcofs = iteratorSrc.GetNextOfs();
+                    entry.dstofs = iteratorDst.GetNextOfs();
+                    assert(entry.srcofs != entry.dstofs);
+                    //entry.delay = true;
+                    entry.stackofs = sArgSrc.m_offs;
+                    pShuffleEntryArray->Append(entry);
+
+                    //entry.srcofs = entry.srcofs;//iteratorSrc.GetNextOfs();
+                    entry.dstofs = iteratorDst.GetNextOfs();
+                    assert(entry.srcofs != entry.dstofs);
+                    //entry.delay = true;
+                    //entry.stackofs = sArgSrc.m_offs;
+                    pShuffleEntryArray->Append(entry);
+
+                    assert(!iteratorSrc.HasNextOfs());
+                    break;//continue;
+                }
+                else if ((sArgSrc.m_offs == 8) || (sArgSrc.m_offs == 7))
+                {//second double.   ||    first double.
+                    assert(sArgDst.m_cGenReg == 1);
+                    assert(sArgDst.m_cFloatReg == 1);
+                    assert(sArgSrc.m_byteStackSize == 16);
+
+                    entry.srcofs = iteratorSrc.GetNextOfs();
+                    entry.dstofs = iteratorDst.GetNextOfs();
+                    assert(entry.srcofs != entry.dstofs);
+                    //entry.delay = true;
+                    entry.stackofs = sArgSrc.m_offs;
+                    pShuffleEntryArray->Append(entry);
+
+                    entry.srcofs = iteratorSrc.GetNextOfs();
+                    entry.dstofs = iteratorDst.GetNextOfs();
+                    assert(entry.srcofs != entry.dstofs);
+                    //entry.delay = true;
+                    //entry.stackofs = sArgSrc.m_offs;
+                    pShuffleEntryArray->Append(entry);
+
+                    assert(!iteratorSrc.HasNextOfs());
+                    break;//continue;
+                }
+
+                assert(!"---------No this stack case.--------------");
+            }
+            //else if (sArgSrc.m_cGenReg > 0)
```

Comment:
Probably this section require some cleanup. Looks a bit in-development.

Reply:
> Probably this section require some cleanup. Looks a bit in-development.

Yes, Thanks for your review again.